### PR TITLE
Add option "doctrine.orm.enable_lazy_ghost_objects"

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Sidekick of https://github.com/doctrine/orm/pull/10187

### CODE REVIEW

**Comment by ostrolucky** [comment]
LGTM, but upstream PR should be merged first

**Comment by stof** [comment]
The upstream PR is merged

**Comment by stof** [comment]
We might want to wait for a release though

**Comment by ostrolucky** [comment]
what about documenting this, at least by ading option description?

File: DependencyInjection/DoctrineExtension.php [hunk 1]
```diff
@@ -489,8 +490,15 @@ protected function ormLoad(array $config, ContainerBuilder $container)
 
         $container->setParameter('doctrine.default_entity_manager', $config['default_entity_manager']);
 
-        $options = ['auto_generate_proxy_classes', 'proxy_dir', 'proxy_namespace'];
+        $options = ['auto_generate_proxy_classes', 'enable_lazy_ghost_objects', 'proxy_dir', 'proxy_namespace'];
         foreach ($options as $key) {
+            if ($key === 'enable_lazy_ghost_objects' && $config[$key] && ! method_exists(OrmConfiguration::class, 'setLazyGhostObjectEnabled')) {
```
Comment (derrabus) [inline DependencyInjection/DoctrineExtension.php]: We should also check if VarExporter and Persistence have been installed in the correct versions.

Comment (nicolas-grekas) [reply DependencyInjection/DoctrineExtension.php]: this is not needed: the checks are already done in setLazyGhostObjectEnabled

**Comment by nicolas-grekas** [comment]
> what about documenting this, at least by ading option description?

I was about to add this but I think it's quite difficult to convey something useful for now.
I'd suggest waiting for doctrine/common to deprecate its proxies before.
And then to deprecate *not* setting the new setting to true later on.

File: DependencyInjection/DoctrineExtension.php (hunk 1 above)
Comment (derrabus) [reply DependencyInjection/DoctrineExtension.php]: Yes, but we would turn a runtime error into a compile-time one.

Comment (nicolas-grekas) [reply DependencyInjection/DoctrineExtension.php]: Good call, PR updated.

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
Upstream PR has been released with ORM 2.14.0.

**Pull request merged**